### PR TITLE
Hotfix coupon service discount

### PR DESCRIPTION
**Pull request by forellana-digithinkit**
--- Added
- Added taxes_and_charges as "services" discounts on coupons.
- Added coupon discount breakdown table on totals areas.

--- Changed
- Fixed bug where shipping method and address would stay on quotation when cart was loaded and UI would not reflect those values. Fix just clears these fields to allow user to reselect.
- Refactored how coupon discounts are recalculated on shipping value changes or resyncing quotation and awc cart.

## DO NOT MERGE UNTIL AFTER the following PRs:

- https://github.com/DigiThinkIT/awesome_cart/pull/278
- https://github.com/DigiThinkIT/awesome_cart/pull/279

coupon-shipping-free

### CODE REVIEW

File: awesome_cart/awc.py [hunk 1]
```diff
@@ -745,6 +747,7 @@ def sync_awc_and_quotation(awc_session, quotation, quotation_is_dirty=False, sav
 	call_awc_sync_hook(awc_session, quotation)
 
 	save_and_commit_quotation(quotation, quotation_is_dirty, awc_session, commit=True, save_session=awc_is_dirty)
+	#calculate_quotation_discount(quotation, awc_session)
```
Comment (Alchez) [inline awesome_cart/awc.py]: Déjà vu. 😅

File: awesome_cart/awesome_cart/doctype/awc_coupon/awc_coupon.json [hunk 2]
```diff
@@ -525,6 +525,66 @@
...
+   "hidden": 0, 
+   "ignore_user_permissions": 0, 
+   "ignore_xss_filter": 0, 
+   "in_filter": 0, 
+   "in_global_search": 0, 
+   "in_list_view": 0, 
+   "in_standard_filter": 0, 
+   "length": 0, 
+   "no_copy": 0, 
+   "options": "AWC Coupon Service",
```
Comment (Alchez) [inline awesome_cart/awesome_cart/doctype/awc_coupon/awc_coupon.json]: When I try to create a coupon, ERPNext errors out and says `DocType AWC Coupon Service not found`.

File: awesome_cart/awc.py [hunk 3]
```diff
@@ -1132,8 +1173,8 @@ def cart(data=None, action=None):
 			quotation.use_customer_fedex_account = 1 if data[0].get(
 				"address", {}).get("use_customer_fedex_account") else 0
 			quotation.flags.ignore_permissions = True
-			quotation.save()
-			frappe.db.commit()
+			#quotation.save()
+			#frappe.db.commit()
```
Comment (Alchez) [inline awesome_cart/awc.py]: Wouldn't this not save the "Use my Fedex account" value?

<sub>I've become wary of the octothorpe now.</sub>

File: awesome_cart/awc.py (hunk 1 above)
Comment (vjFaLk) [reply awesome_cart/awc.py]: Not the same as `calculate_tax_and_totals` btw

File: awesome_cart/awc.py (hunk 3 above)
Comment (vjFaLk) [reply awesome_cart/awc.py]: 🐙

File: awesome_cart/awc.py (hunk 1 above)
Comment (forellana-digithinkit) [reply awesome_cart/awc.py]: haha, yeah not the same. I moved that into the save_and_commit_quotation() method so its ubiquitous during calcs.

File: awesome_cart/awesome_cart/doctype/awc_coupon/awc_coupon.json (hunk 2 above)
Comment (forellana-digithinkit) [reply awesome_cart/awesome_cart/doctype/awc_coupon/awc_coupon.json]: Mmmm... will check if the doctype made it to repo

File: awesome_cart/awc.py (hunk 3 above)
Comment (forellana-digithinkit) [reply awesome_cart/awc.py]: If you notice, save and commit are already happening on line 1248. Why save and commit twice on the same call?

Comment (Alchez) [reply awesome_cart/awc.py]: But that save and commit is for when the `action` is "updateItem", whereas this one is for "calculate_shipping", and returns on line 1181. It'll give you the correct rates either way, but I'll confirm if the "Use Customer Fedex Account" check is properly applied.

Comment (forellana-digithinkit) [reply awesome_cart/awc.py]: Ah sorry, was looking at the wrong thing. I meant 1179. So right before the return there is call to calc shipping and we are indeed saving and committing before returning from that.

Comment (Alchez) [reply awesome_cart/awc.py]: Ohh, I see. No problem, then. 👍😄

**Comment by forellana-digithinkit** [comment]
@Alchez Just added the missing doctype, had it as "custom?" by mistake

**Comment by vjFaLk** [comment]
@forellana-digithinkit pls fix

**Comment by vjFaLk** [comment]
I looked at AwesomeCartJS to see if I can generate the standalone.js file... 

I have no idea what I'm doing.

**Pull request merged**